includes/Embed_Block.php

```diff
@@ -111,6 +108,87 @@ public function init() {
 		add_filter( 'wp_kses_allowed_html', [ $this, 'filter_kses_allowed_html' ], 10, 2 );
 	}
 
+	/**
+	 * Initializes assets needed for the block.
+	 *
+	 * Registers scripts and styles for the block itself used in the editor,
+	 * as well as the external assets used by the <amp-story-player> web component (non-AMP version).
+	 *
+	 * Also registers an inline stylesheet used on the frontend to ensure correct
+	 * styling and responsiveness for the block, which is used for both the regular and the AMP version.
+	 *
+	 * This CSS is added as inline style as we don't yet have CSS processing in place to have this as an external stylesheet.
+	 * Plus, it saves an extra network request.
+	 *
+	 * @todo Consider moving inline CSS to external stylesheet.
+	 * @todo Consider moving style registration to third class.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return void
+	 */
+	private function init_assets() {
+		wp_register_script( 'standalone-amp-story-player', 'https://cdn.ampproject.org/amp-story-player-v0.js', [], 'v0', false );
+		wp_register_style( 'standalone-amp-story-player', 'https://cdn.ampproject.org/amp-story-player-v0.css', [], 'v0' );
+
+		$this->register_script( self::SCRIPT_HANDLE, [ 'standalone-amp-story-player', Tracking::SCRIPT_HANDLE ] );
+		$this->register_style( self::SCRIPT_HANDLE, [ 'standalone-amp-story-player' ] );
+
+		// Registering a style without a `src` allows us to just use the inline style below
+		// without needing an external stylesheet.
+		wp_register_style(
+			self::SCRIPT_HANDLE_FRONTEND,
+			'',
+			[],
+			WEBSTORIES_VERSION
+		);
+
+		$css = <<<CSS
```

Comment:
@swissspidy Does this CSS snippet go through the css linter? Just wonder. If not, I believe it should. Maybe it should be write in a CSS file and loaded into memory?

Reply:
I'd rather actually load this as an external stylesheet instead of doing `file_get_contents` or the like here.

Reply:
@spacedmonkey thoughts?

Reply:
Isn't there a performance reason you didn't do that in the first place. For the record, I don't love using `file_get_contents` here, but is this best for the user.

Reply:
How is using file_get_contents better for the user?

I will change it, but just using stylelint is not much added value. Ideally we‘d minify CSS etc.

Reply:
See also how this is being done in the AMP plugin:

https://github.com/ampproject/amp-wp/blob/7a41a7b3bb06972ca73cfec8dbe693635cc0499d/src/MobileRedirection.php#L418-L419

It can pull in the built CSS file rather than the source one.